### PR TITLE
[14.0] hr_holidays_public: add public holidays as unusual days

### PR DESCRIPTION
**Pull request by petrus-v**
in order to make a grey/disabled date in the calendar view

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Public holidays may depend on the user and their location (country and state), so we should restrict it. It's the call to this method linked to one user? Now that currently a calendar event is created for the public holiday.

**Comment by petrus-v** [comment]
> Public holidays may depend on the user and their location (country and state), so we should restrict it. It's the call to this method linked to one user? Now that currently a calendar event is created for the public holiday.

Good catch !

[hr holliday module consider current user](https://github.com/OCA/OCB/blob/14.0/addons/hr_holidays/models/hr_leave.py#L1257) to display unusual days (like weekends according calendar resource model): for a manager that would validate holidays to an employee it will get it's own unusual days (not the one from the employee that he is checking).

So what i can do is similar: filter public holidays by country of the current user according country define on self.env.user.employee_id.address_id.country_id (with fallback to the current company country) if set assuming manager live in the same country as subordinates and add a warning with this limitation in readme.

**Comment by pedrobaeza** [comment]
Great, we have a plan :wink:

File: hr_holidays_public/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -1,4 +1,6 @@
-This module is a technical module to handle public holidays.
+This module handle public holidays.
```
Comment (pedrobaeza) [inline hr_holidays_public/readme/DESCRIPTION.rst]:
```suggestion
This module handles public holidays.
```

File: hr_holidays_public/models/hr_leave.py [hunk 2]
```diff
@@ -18,3 +18,31 @@ def _get_number_of_days(self, date_from, date_to, employee_id):
...
+                (
+                    "date",
+                    "<",
+                    date_to,
+                )
+            )
+        country_id = self.env.user.employee_id.address_id.country_id.id
+        if not country_id:
+            country_id = self.env.company.country_id.id or False
+        if country_id:
```
Comment (pedrobaeza) [inline hr_holidays_public/models/hr_leave.py:37]: Something similar should be done for the states.

**Review by pedrobaeza — APPROVED** [review]

**Review by AntoniRomera — APPROVED** [review]

**Comment by AntoniRomera** [comment]
I see this as a good addition, good work.

**Pull request merged**